src/main/java/com/xceptance/neodymium/util/JsonAssert.java

```diff
@@ -0,0 +1,146 @@
+package com.xceptance.neodymium.util;
+
+import org.json.JSONException;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import io.qameta.allure.Allure;
+
+public class JsonAssert
+{
+    private static String htmlScript(String expectedJson, String actualJson, JSONCompareMode jsonCompareMode) 
+    {
+        boolean arrayOrderDoesntMatters = false;
+        boolean arrayIsExtensible = false;
+        
+        if (jsonCompareMode == JSONCompareMode.LENIENT) 
+        {
+            arrayOrderDoesntMatters = true;
+            arrayIsExtensible = true;
+        }
+        else if (jsonCompareMode == JSONCompareMode.NON_EXTENSIBLE) 
+        {
+            arrayOrderDoesntMatters = true;
+        }
+        
+        String html = ""
+            + "<html>"
+              + "<head>"
+                + "<link"
+                  + " rel=\"stylesheet\""
+                  + " href=\"https://esm.sh/jsondiffpatch@0.6.0/lib/formatters/styles/html.css\""
+                  + " type=\"text/css\""
+                + "/>"
+              + "</head>"
+              + "<body>"
+                + "<div id=\"visual\"></div>"
+                + "<script type=\"module\">"
+                  + "import * as jsondiffpatch from 'https://esm.sh/jsondiffpatch@0.6.0';"
+                  + "import * as htmlFormatter from 'https://esm.sh/jsondiffpatch@0.6.0/formatters/html';"
+                  
+                  + "const detectMoveFalseInstance = jsondiffpatch.create({"
+                    + "arrays: {"
+                      + "detectMove: false"
+                    + "}"
+                  + "});"
+                  + "const detectMoveTrueInstance = jsondiffpatch.create({"
+                    + "arrays: {"
+                      + "detectMove: true"
+                    + "}"
+                  + "});"
+
+                  + "const left = " + expectedJson + ";"
+                  + "const right = " + actualJson + ";"
+                  
+                  + "function sortArraysInJson(json) {"
+                    + "for (const key in json) {"
+                      + "if (Array.isArray(json[key])) {"
```

Comment:
this crashes when an empty array is part of the json. please change to `if (Array.isArray(json[key]) && json[key].lenght>0) {`

Reply:
done (ist aber .length ;) )